analyzers/tests/SonarAnalyzer.UnitTest/TestCases/RedundantConditionalAroundAssignment.CSharp9.cs

```diff
@@ -3,14 +3,57 @@
 
 y = y switch
 {
-    not 5 => 5 // Compliant - FN
+    not 5 => 5 // Noncompliant
 };
 
 y = y switch
```

Comment:
I would say this is an edge case.  Because y will be set to 5 only if it was 5, otherwise an exception will be thrown.

Reply:
No action